swift/apple/FirezoneKit/Sources/FirezoneKit/Stores/AuthStore.swift

```diff
@@ -60,8 +76,50 @@ final class AuthStore: ObservableObject {
     tunnelStore.$tunnelAuthStatus
       .sink { [weak self] tunnelAuthStatus in
         guard let self = self else { return }
+        logger.log("Tunnel auth status changed to: \(tunnelAuthStatus)")
         Task {
-          self.loginStatus = await self.getLoginStatus(from: tunnelAuthStatus)
+          let loginStatus = await self.getLoginStatus(from: tunnelAuthStatus)
+          if tunnelStore.tunnelAuthStatus == tunnelAuthStatus {
+            // Make sure the tunnelAuthStatus hasn't changed while we were getting the login status
+            self.loginStatus = loginStatus
+          }
+        }
+      }
+      .store(in: &cancellables)
+
+    tunnelStore.$status
+      .sink { [weak self] status in
+        guard let self = self else { return }
+        Task {
+          if status == .disconnected {
+            self.logger.log("\(#function): Disconnected")
+            if let tsEvent = TunnelShutdownEvent.loadFromDisk() {
+              self.logger.log(
+                "\(#function): Tunnel shutdown event: \(tsEvent, privacy: .public)"
+              )
+              switch tsEvent.action {
+              case .signoutImmediately:
+                self.signOut()
+              case .retryThenSignout:
+                let shouldReconnect = (self.reconnectionAttemptsRemaining > 0)
+                self.reconnectionAttemptsRemaining = self.reconnectionAttemptsRemaining - 1
+                if shouldReconnect {
+                  self.logger.log(
+                    "\(#function): Trying to reconnect (\(self.reconnectionAttemptsRemaining) attempts after this)"
+                  )
```

Comment:
Would it make sense to sleep for a second in between attempts?

Reply:
Sure, done.